### PR TITLE
Fix hyphen escaping in resource paths for agent tool creation from connectors

### PR DESCRIPTION
**Pull request by dan-niles**
## Purpose

Fixes https://github.com/wso2/product-integrator/issues/1233

This PR fixes hyphen escaping in resource paths for agent tools generated from connectors.

https://github.com/user-attachments/assets/c13de57c-f3bb-465a-8f56-fc3f6724d0bd

- [ ] Added reusable UI components to the ui-toolkit. Follow the [intructions](./workspaces/common-libs/ui-toolkit/README.md) when adding the componenent.
- [ ] Use ui-toolkit components wherever possible. Run `npm run storybook` from the root directory to view current components.
- [ ] Matches with the native VSCode look and feel.

- [ ] Added Icons to the font-wso2-vscode. Follow the [instructions](./workspaces/common-libs/font-wso2-vscode/README.md).

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

 

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved selection handling in the AI Chat Agent panel to prevent conflicts from rapid interactions
  * Enhanced form submission reliability with better state management during asynchronous operations
  * Fixed UI loading state management to ensure proper cleanup after field operations

* **New Features**
  * Added a reusable helper to reliably update resource path templating during form updates

* **Tests**
  * Added unit tests covering path replacement, edge cases, and non-destructive behavior

### CODE REVIEW

**Comment by dan-niles** [comment]
@CodeRabbit review

File: workspaces/ballerina/ballerina-visualizer/src/views/BI/AIChatAgent/AIAgentSidePanel.tsx [hunk 1]
```diff
@@ -717,17 +727,17 @@ export function AIAgentSidePanel(props: BIFlowDiagramProps) {
...
-                                codedata: resourcePathProperty.codedata ? {
+                                codedata: {
                                     ...resourcePathProperty.codedata,
-                                    originalName: typeof updatedPath === "string" ? updatedPath : String(updatedPath)
-                                } : {
-                                    originalName: typeof updatedPath === "string" ? updatedPath : String(updatedPath)
+                                    originalName: String(replacePathParam(sourcePath))
                                 },
-                                value: updatedPath
+                                value: replacePathParam(displayPath)
```
[the remark this answers is not in the document]
Comment (dan-niles) [reply workspaces/ballerina/ballerina-visualizer/src/views/BI/AIChatAgent/AIAgentSidePanel.tsx]: fixed

File: workspaces/ballerina/ballerina-visualizer/src/views/BI/AIChatAgent/AIAgentSidePanel.tsx [hunk 2]
```diff
@@ -717,17 +727,17 @@ export function AIAgentSidePanel(props: BIFlowDiagramProps) {
...
                             };
                         }
                     }
```
[the remark this answers is not in the document]
Comment (dan-niles) [reply workspaces/ballerina/ballerina-visualizer/src/views/BI/AIChatAgent/AIAgentSidePanel.tsx:737]: 1. Coercion: `resourcePath.value` is always a string for `RESOURCE_ACTION_CALL` nodes — the `String()` wrapper is just to satisfy TS (`CodeData.originalName` is typed as string). The asymmetry doesn't manifest in practice.
2. Per-iteration recreation: runs once per form submit over a handful of keys, hence the performance impact is negligible.
3. Global replace: Not possible in ballerina, to have duplicate identifiers in the resource function path parameters: `resource function post foo/[string id]/[string id]()`

**Review by kanushka — APPROVED** [review]

**Review by kanushka — APPROVED** [review]
